addons/mrp/mrp.py

```diff
@@ -322,14 +333,8 @@ def _factor(factor, product_efficiency, product_rounding):
 
             #If BoM should not behave like PhantoM, just add the product, otherwise explode further
             if bom_line_id.type != "phantom" and (not bom_id or self.browse(cr, uid, bom_id, context=context).type != "phantom"):
-                result.append({
-                    'name': bom_line_id.product_id.name,
-                    'product_id': bom_line_id.product_id.id,
-                    'product_qty': quantity,
-                    'product_uom': bom_line_id.product_uom.id,
-                    'product_uos_qty': bom_line_id.product_uos and _factor(bom_line_id.product_uos_qty * factor, bom_line_id.product_efficiency, bom_line_id.product_rounding) or False,
-                    'product_uos': bom_line_id.product_uos and bom_line_id.product_uos.id or False,
-                })
+                product_uos_qty = bom_line_id.product_uos and _factor(bom_line_id.product_uos_qty * factor, bom_line_id.product_efficiency, bom_line_id.product_rounding) or False
```

Comment:
I understand that the method _factor is only accessible in this method as it's declared here. This forbid you to use it in the _prepare method. Maybe the best solution is to declare this method for the full file at the top of mrp.py before declaring the first class? so you can use it in the prepare method